js/dashicons-picker.js

```diff
@@ -0,0 +1,422 @@
+			'pets',
+			'privacy',
+			'superhero',
+			'superhero-alt',
+			'edit-page',
+			'fullscreen-alt',
+			'fullscreen-exit-alt',
+			'',
+			'',
+			''
+		];
+
+		return this.each( function () {
+
+			var button = $( this ),
+				offsetTop,
+				offsetLeft;
+
+			button.on( 'click.dashiconsPicker', function ( e ) {
+				offsetTop = $( e.currentTarget ).offset().top;
+				offsetLeft = $( e.currentTarget ).offset().left;
+				createPopup( button );
+			} );
+
+			function createPopup( button ) {
+
+				var target = $( '#menu_icon' ),
+					preview = $( button.data( 'preview' ) ),
+					popup  = $( '<div class="dashicon-picker-container">' +
+						'<div class="dashicon-picker-control"></div>' +
+						'<ul class="dashicon-picker-list"></ul>' +
+					'</div>' ).css( {
+						'top':  offsetTop,
+						'left': offsetLeft
+					} ),
+					list = popup.find( '.dashicon-picker-list' );
+
+				for ( var i in icons ) {
+					if ( icons.hasOwnProperty(i) ) {
+						list.append('<li data-icon="' + icons[i] + '"><a href="#" title="' + icons[i] + '"><span class="dashicons dashicons-' + icons[i] + '"></span></a></li>');
+					}
+				}
+
+				$( 'a', list ).on( 'click', function ( e ) {
+					e.preventDefault();
+					var title = $( this ).attr( 'title' );
+					target.val( 'dashicons-' + title );
+					preview
+						.prop('class', 'dashicons')
+						.addClass( 'dashicons-' + title );
+					removePopup();
+				} );
+
+				var control = popup.find( '.dashicon-picker-control' );
+
+				control.html( '<a data-direction="back" href="#">' +
+					'<span class="dashicons dashicons-arrow-left-alt2"></span></a>' +
+					'<input type="text" class="" placeholder="Search" />' +
+					'<a data-direction="forward" href="#"><span class="dashicons dashicons-arrow-right-alt2"></span></a>'
+				);
+
+				$( 'a', control ).on( 'click', function ( e ) {
+					e.preventDefault();
+					if ( $( this ).data( 'direction' ) === 'back' ) {
+						$( 'li:gt(' + ( icons.length - 26 ) + ')', list ).prependTo( list );
+					} else {
+						$( 'li:lt(25)', list ).appendTo( list );
+					}
+				} );
+
+				popup.appendTo( 'body' ).show();
+
+				$( 'input', control ).on( 'keyup', function ( e ) {
+					var search = $( this ).val();
+					if ( search === '' ) {
+						$( 'li:lt(25)', list ).show();
+					} else {
+						$( 'li', list ).each( function () {
+							if ( $( this ).data( 'icon' ).toLowerCase().indexOf( search.toLowerCase() ) !== -1 ) {
+								$( this ).show();
+							} else {
+								$( this ).hide();
+							}
+						} );
+					}
+				} );
+
+				$( document ).on( 'mouseup.dashicons-picker', function ( e ) {
+					if ( ! popup.is( e.target ) && popup.has( e.target ).length === 0 ) {
+						removePopup();
+					}
+				} );
+			}
+
+			function removePopup() {
+				$( '.dashicon-picker-container' ).remove();
+				$( document ).off( '.dashicons-picker' );
+			}
+		} );
+	};
+
+	$( function () {
+		$( '.dashicons-picker' ).dashiconsPicker();
+	} );
+
+}( jQuery ) );
```

Comment:
Anything that can be done to also provide a minified version of this file? Something needing edited on any build process on the CPTUI side with any tools we already have?

Reply:
I added minified version also.